src/components/UserRegister.tsx

```diff
@@ -1,84 +1,138 @@
-import {TextField} from "@mui/material";
+import { TextField } from "@mui/material";
 import Box from "@mui/material/Box";
-import React, {useContext, useState} from "react";
+import React, { useContext, useState } from "react";
 import Button from "@mui/material/Button";
-import LoadingButton from '@mui/lab/LoadingButton';
-import SendIcon from '@mui/icons-material/Send';
+import LoadingButton from "@mui/lab/LoadingButton";
+import SendIcon from "@mui/icons-material/Send";
+import Alert from "@mui/material/Alert";
 import axios from "axios";
 import { RegisterContext } from "../contexts/RegisterContext";
+import { registerSchema } from "../validations/RegisterValidations";
 
-
-const Register = ({onButtonClick}) =>{
-
-  const [ processing, setProcessing] = useState(false);
+interface registerForm {
+  name: "";
+  email: "";
+  password: "";
+  confirmPassword: "";
+}
+const Register = ({ onButtonClick }) => {
+  const [processing, setProcessing] = useState(false);
   const { registerForm, modifyForm } = useContext(RegisterContext);
-
+  const [errors, setErrors] = useState({
+    name: "",
+    email: "",
+    password: "",
+    confirmPassword: "",
+  });
```

Comment:
I have implemented the form struct in `RegisterContext.tsx`, maybe we can reuse it rather than maintaining the form inside the component.